aten/src/ATen/core/Half-inl.h

```diff
@@ -188,6 +188,137 @@ inline AT_HOSTDEVICE Half operator/(int a, Half b) {
 } // namespace at
 
 namespace std {
+  
+// specializing std for common mathematical functions to be used with at::Half.
+// For most of this functions the cmath header/math_functions.hpp in cuda 
+// would call the corresponding float specialization. 
+// However, here we are calling the float functions
+// directly to save one API call, like it's been done in the old THCNumerics.cuh
+// Also note that some of these functions might not have been used in the library
+// at all. However, keeping all the function definitions to provide backward
+// compatibility
+
+inline AT_HOSTDEVICE at::Half lgamma(at::Half a) {
```

Comment:
I don't think we should add these overloads. With a few exceptions, it's undefined behavior to add declarations to namespace `std`. (This does not meet any of the exceptions)

https://en.cppreference.com/w/cpp/language/extending_std